Cyotek.Windows.Forms.ImageBox/ImageBox.cs

```diff
@@ -4062,21 +4098,63 @@ protected override void OnMouseWheel(MouseEventArgs e)
     {
       base.OnMouseWheel(e);
 
-      if (this.AllowZoom && this.SizeMode == ImageBoxSizeMode.Normal)
+      void DoZoom()
       {
-        int spins;
+        if (this.AllowZoom && this.SizeMode == ImageBoxSizeMode.Normal)
+        {
+          int spins;
 
-        // The MouseWheel event can contain multiple "spins" of the wheel so we need to adjust accordingly
-        spins = Math.Abs(e.Delta / SystemInformation.MouseWheelScrollDelta);
+          // The MouseWheel event can contain multiple "spins" of the wheel so we need to adjust accordingly
+          spins = Math.Abs(e.Delta / SystemInformation.MouseWheelScrollDelta);
 
-        // TODO: Really should update the source method to handle multiple increments rather than calling it multiple times
-        for (int i = 0; i < spins; i++)
+          // TODO: Really should update the source method to handle multiple increments rather than calling it multiple times
+          for (int i = 0; i < spins; i++)
+          {
+            this.ProcessMouseZoom(e.Delta > 0, e.Location);
+          }
+        }
+      }
+
+      if (MouseWheelMode == ImageBoxMouseWheelMode.Zoom)
+      {
+        DoZoom();
+      }
+      else if (MouseWheelMode == ImageBoxMouseWheelMode.ScrollAndZoom)
+      {
+        if (ModifierKeys == Keys.Control)
+        {
+          DoZoom();
+        }
+        else if (VerticalScroll.Visible && VerticalScroll.Enabled && ModifierKeys == Keys.None)
```

Comment:
Unfortunately, the `VerticalScroll` and `HorizontalScroll` properties are essentially broken and don't do what you think they will do (there is a branch where I'm (oh so slowly) redoing the control to remove this completely). The `Visible` and `Enabled` properties always return `true` - you can see in this in action by having a zoomed out image (e.g. no scrollbars), then holding shift and spinning the mouse wheel in reverse - the image will actually shrink because behind the scenes it is scrolling as requested.

You would need to switch to use the `VScroll` and `HScroll` properties which will then work as expected.

Reply:
Great to have such a detailed review. I will change this also.